scripts/create_gw5000_database.py

```diff
@@ -0,0 +1,142 @@
+    source_download_link = '# Source download link: https://dataserv.ub.tum.de/index.php/s/m1507656 , Filename: df_5k.json'
+    source_method = '# Source method: PBE_TS-vdW'
+    date_added = f'# Date added: 2023-02-05  14:14:48 ({author})'
+    date_modified = f'# Date modified: {datetime.today()} ({author})'
+    common_name = '# Common name: Unknown'
+    smarts = '# Smarts: Unknown'
+    cas = '# CAS: Unknown'
+    spin_multiplicity = 'Singlet'
+    electronic_state = 'Ground'
+    charge = 0
+    point_group = 'Unknown'
+    spm_info = f'Spin multiplicity = {spin_multiplicity}'
+    electron_state_info = f'Electronic state = {electronic_state}'
+    charge_info = f'Charge = {charge}'
+    point_group_info = f'Point group = {point_group}'
+
+    molecule_names = []
+    output_info = []
+
+    inchi_col_vals = df_5k.inchi.values
+    smiles_col_vals = df_5k.canonical_smiles.map(lambda x: x.strip('\n').strip('\t'))
+    xyz_col_vals = df_5k.xyz_pbe_relaxed.map(lambda x: remove_xyz_header(x))
+    n_atoms_col_vals = df_5k.number_of_atoms
+    csd_code_col_vals = df_5k.refcode_csd
+
+
+    for inchi,smiles, xyz, atom_count, csd_code in tqdm(
+            zip(inchi_col_vals,smiles_col_vals,xyz_col_vals, n_atoms_col_vals, csd_code_col_vals),
+            desc='Generating gw5k dataset: ', total=df_5k.shape[0]):
+
+        molecular_formula = convert_smiles_to_molecular_formula(inchi)
+        n_electrons_in_mol = calculate_n_electrons_in_mol(xyz)
+        n_elec_info = f'Number of electrons = {n_electrons_in_mol}'
+        properties = f'# Properties: {n_elec_info}, {spm_info}, {electron_state_info}, {charge_info}, {point_group_info}'
+        meta_data_comments = f'{source_paper}\n{source_data_host}\n{source_download_link}\n{source_method}\n{properties}\n{date_added}\n{date_modified}\n'
+        reference_code = f'# Reference code: {csd_code}'
+
+        if molecular_formula in molecule_names:
+            n_times_molecule = molecule_names.count(molecular_formula)
+            name = f'NAME = {molecular_formula}({n_times_molecule + 1}):GW5000.v{version}'
+        else:
+            name = f'NAME = {molecular_formula}:GW5000.v{version}'
+
+        molecule_names.append(molecular_formula)
+
+        smiles_info = f'# SMILES : {smiles}'
+        n_atoms = f'# Number of atoms: {atom_count}'
+        inchi_info = f'# {inchi}'
+        molecule_info = f'{name}\n{n_atoms}\n{common_name}\n{inchi_info}\n{smiles_info}\n\n{smarts}\n\n{reference_code}\n{cas}\n{meta_data_comments}\n{xyz}\n'
+
+        output_info.append(molecule_info)
+
+    with open(out_geom_database_filepath, "w") as text_file:
+        for mol_str in output_info:
+            text_file.write(mol_str)
+
+    e_time = datetime.now(tz=tz.gettz('Europe/Stockholm'))
+    print(f'Endded at time : {e_time}')
+    print(f'time diff: {e_time - s_time}')
+
+
+if __name__ == '__main__':
+    output_geom_database_filepath = "gw5000.txt"
```

Comment:
Can the output filename be set in a place that's closer to (or immediately next to) the information like "version number" and "URL link" and "name of contributor"? This way when we use this script for the ~61,000 molecule database, or a different database, all of the things that have to be changed "manually" can be done so together, without as much of a risk of missing anything.

Reply:
Done.